velox/functions/prestosql/aggregates/tests/GeometricMeanTest.cpp

```diff
@@ -0,0 +1,230 @@
+          [](auto row) {
+            // All values in group 3 are null.
+            if (row / 10 == 3) {
+              return true;
+            }
+
+            // Every other value is null.
+            return row % 2 == 0;
+          }),
+  });
+
+  auto expected = makeRowVector({
+      makeFlatVector<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}),
+      makeFlatVector<double>(
+          10,
+          [](auto row) {
+            double logSum = 0;
+            int64_t count = 0;
+            for (int32_t i = 1; i < 10; i += 2) {
+              logSum += std::log(row * 10 + i);
+              count++;
+            }
+            return std::exp(logSum / count);
+          },
+          [](auto row) { return row == 3; }),
+  });
+
+  testAggregations({data}, {"c0"}, {"geometric_mean(c1)"}, {expected});
+}
+
+TEST_F(GeometricMeanTest, groupByIntegers) {
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(100, [](auto row) { return row / 10; }),
+      makeFlatVector<int64_t>(100, [](auto row) { return row; }),
+  });
+
+  auto expected = makeRowVector({
+      makeFlatVector<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}),
+      makeFlatVector<double>(
+          10,
+          [](auto row) {
+            double logSum = 0;
+            int64_t count = 0;
+            for (int32_t i = 0; i < 10; ++i) {
+              logSum += std::log(row * 10 + i);
+              count++;
+            }
+            return std::exp(logSum / count);
+          }),
+  });
+
+  testAggregations({data}, {"c0"}, {"geometric_mean(c1)"}, {expected});
+}
+
+TEST_F(GeometricMeanTest, globalDoubles) {
+  auto data = makeRowVector({
+      makeFlatVector<double>(100, [](auto row) { return row * 0.1 / 7; }),
+  });
+
+  double logSum = 0;
+  int64_t count = 0;
+  for (int32_t i = 0; i < 100; ++i) {
+    logSum += std::log(i * 0.1 / 7);
+    count++;
+  }
+  double geometricMean = std::exp(logSum / count);
+
+  auto expected = makeRowVector({
+      makeFlatVector(std::vector<double>{
+          geometricMean,
+      }),
+  });
+
+  testAggregations({data}, {}, {"geometric_mean(c0)"}, {expected});
+}
+
+TEST_F(GeometricMeanTest, groupByDoubles) {
+  auto data = makeRowVector({
+      makeFlatVector<int32_t>(100, [](auto row) { return row / 10; }),
+      makeFlatVector<double>(100, [](auto row) { return row * 0.1; }),
+  });
+
+  auto expected = makeRowVector({
+      makeFlatVector<int32_t>({0, 1, 2, 3, 4, 5, 6, 7, 8, 9}),
+      makeFlatVector<double>(
+          10,
+          [](auto row) {
+            double logSum = 0;
+            int64_t count = 0;
+            for (int32_t i = 0; i < 10; ++i) {
+              logSum += std::log(row + i * 0.1);
+              count++;
+            }
+            return std::exp(logSum / count);
+          }),
+  });
+
+  testAggregations({data}, {"c0"}, {"geometric_mean(c1)"}, {expected});
+}
+
+TEST_F(GeometricMeanTest, groupByTwoPhases) {
+  // Use two data vectors to test two-phase agg
```

Comment:
> groupByTwoPhases

Usage of "two phases" term here is confusing. testAggregation runs lots of plans including two phase plan partial + final. 

Here, you are running aggregation on multiple batches / vectors of inputs. Hence, let's rename to groupByMultipleBatches.

Reply:
done